julearn/pipeline.py

```diff
@@ -180,25 +176,18 @@ def transform(self, X):
                        )
         return X_trans
 
-    def fit_transform_confounds(self, X, y):
-        if self.confounds is None:
-            return X
-        else:
-            return self.confound_dataframe_pipeline.fit_transform(X, y)
+    def transform_target(self, X, y):
+        X_conf_trans = self.transform_confounds(X)
+        y_true = self._transform_target(X_conf_trans, y)
+        return y_true
 
     def transform_confounds(self, X):
+        X = self._recode_columns(X)
         if self.confounds is None:
             return X
         else:
             return self.confound_dataframe_pipeline.transform(X)
 
-    def transform_target(self, X, y):
-        if self.y_transformer is not None:
-            y_true = self.y_transformer.transform(X, y)
-        else:
-            y_true = y
-        return y_true
-
     def preprocess(self, X, y):
         X_conf_trans = self.transform_confounds(X)
         y_trans = self.transform_target(X_conf_trans, y)
```

Comment:
The aim of the refactor was to make `transform_target` only use the X instead of the X_conf_trans as discussed in #17 
So, it is not a bug in scorer. Now, `transform_target` transforms the confounds internally.
But I've missed changing it in preprocess as well.